### ISSUE TITLE
NoSymbol has no owner

### ISSUE THREAD

**Issue by char101** [harrah/browse#16]
Hi,

I got nosymbol has no owner with this code from play 2.2.0 source

``` scala
package play.api.libs.iteratee {

  private[iteratee] object internal {
    import play.api.libs.iteratee.Iteratee
    import scala.concurrent.{ ExecutionContext, Future }
    import scala.util.control.NonFatal
```

When running fullNameString (Browse.scala:470) on `internal`, the compiler raises the nosymbol error.

**Comment by harrah** [comment]
Thanks, a self-contained test case would be great.

**Comment by rorygraves** [comment]
Hi Mark,

``` scala
object Failure {
  def foo[E](x : Int) = {
    case class Result(i : Int)
    Result(x)
  }
}
```

Triggers the error Browse.normalise in the `if (sym.isCaseApplyOrUnapply)` results - NoSymbol when dereferencing `sym.owner.companionClass` later in the code.

```
sym: method apply
sym.owner: object Result
sym.owner.companionClass: <none>
sym.fullName: Failure:Result:apply
```

I replaced the `sym.owner.companionClass` with 

```
val cclass = sym.owner.companionClass
if (cclass == NoSymbol) sym else cclass
```

This appears to work but I'm not sure whether this is the correct solution (I can also generate the sxr html for the play framework).

Let me know if this change looks good (or suggest an alternate) and I will generate a PR.

Cheers

Rory

**Comment by harrah** [comment]
Thanks, Rory.  Your fix looks good.  I think returning `sym.owner` instead of `sym` might be correct- the intent is for the `Result` in the call `Result(x)` to link to the `Result` in `case class Result(i: Int)`.  Whichever one of `sym` or `sym.owner` does that would be the right one to return.

**Comment by rorygraves** [comment]
I will check the result of `sym` and `sym.owner` and will generate a PR with the right one.
Thanks for your comments on my other code btw.

**Comment by rorygraves** [comment]
@harrah  I added this case 

```
  // test case class usage within a method
  def foo(x : Int) = {
    case class Result(i : Int)
    Result(x)
  }
```

to `NestedDefs.scala`

The difference in the results of is only in the href generated for `i` 
For `sym` it is `<a title="Int" id="NestedDefs.foo;Result.apply.i">i</a>`
For `sym.owner` it is `<a title="Int" id="NestedDefs.foo;Result.i">i</a>` 

As this is a case class I would lean towards `sym.owner`` - what are your thoughts?

`Result(x)` always generates as `(x:Int)Result` - I guess this would be a different fix (to return `Result(x)`

Cheers

Rory

**Comment by harrah** [comment]
Yes, I agree `sym.owner` looks best.  Something to try further is to reference `i` within the body of the Result class.  Whatever anchor that references is probably the right way to go (I assume it will be the `Result.i` one).

Since `Result(x)` is a constructor call, `(Int)Result` seems ok to me.  What do you think is wrong here?

**Comment by rorygraves** [comment]
Pull request as above - Ignore the comment on `Result(x)` I was being daft and misreading it.
